pkg/descheduler/evictions/evictions.go

```diff
@@ -230,21 +215,26 @@ func WithLabelSelector(labelSelector labels.Selector) func(opts *Options) {
 
 type constraint func(pod *v1.Pod) error
 
-type evictable struct {
+type EvictorFilter struct {
 	constraints []constraint
 }
 
-// Evictable provides an implementation of IsEvictable(IsEvictable(pod *v1.Pod) bool).
-// The method accepts a list of options which allow to extend constraints
-// which decides when a pod is considered evictable.
-func (pe *PodEvictor) Evictable(opts ...func(opts *Options)) *evictable {
+func NewEvictorFilter(
+	nodes []*v1.Node,
+	nodeIndexer podutil.GetPodsAssignedToNodeFunc,
+	evictLocalStoragePods bool,
+	evictSystemCriticalPods bool,
+	ignorePvcPods bool,
+	evictFailedBarePods bool,
```

Comment:
as we move forward, I'd like to think about how we can generalize passing/configuring all of these options for better customization.

Reply:
The goal is to transform the `NewEvictorFilter` into `DefaultEvictor` plugin (the name can be different at the end). Making `evictLocalStoragePods`, `evictSystemCriticalPods`, `ignorePvcPods`, `evictFailedBarePods` (and possible other) as arguments to the plugin.